src/components/tab/bl-tab.ts

```diff
@@ -0,0 +1,64 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import style from './bl-tab.css';
+
+@customElement('bl-tab')
+export default class BlTab extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query('.container') private tab: HTMLDivElement;
+
+  @property({ type: String })
+  title: string;
+
+  @property({ type: String })
+  caption: string;
+
+  @property({ type: String, reflect: true })
+  panel: string;
+
+  @property({ type: String, reflect: true })
+  name: string;
+
+  @property({ type: Boolean, reflect: true })
+  selected: boolean;
+
+  handleClick(e: Event) {
+    const detail = { panel: this.panel, tab: this.tab };
+    const event = new CustomEvent('tabClicked', { detail, bubbles: true, composed: true, cancelable: true });
+    this.dispatchEvent(event);
```

Comment:
Please use `@event` decorator.

Reply:
We should share the event between the tab component in a slot and its parent, so there is not any way to use the ` @event `  decorator. Because we can not catch events from outside the slot.

Reply:
`@event` decorator also just calls `dispatchEvent`. So technically there should be no difference.